### PR TITLE
Add augmented assignment proposal

### PR DESCRIPTION
**Pull request by Leonard-Reuter**
Added proposal for augmented assignment with `+.=`, `/.=` etc. following the discussion in #113

### CODE REVIEW

**Comment by certik** [comment]
Thanks for writing it up! 

In order for this proposal to have a chance, in my opinion we have to address existing and past committee criticisms.

You should reference previous proposals, I think there were some.

And then we need to go into the J3 mailinglist (I added a link in one of the comments in #113), and address every objection. Otherwise the same objections will be given and the proposal rejected.

File: augmented_assignment/augmented_assignment.txt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    real :: A(n)
+    ...
+    A(get_index()) +.= x
+
+If the above example were to be written without augmented assigment, how
+often `get_index` is called would depend on the compiler.
+
+The dot is added between the operator and `=`, since `/=` is already in
+use.
```
Comment (ivan-pi) [inline augmented_assignment/augmented_assignment.txt:41]: Here it would be good to add "in use as the *not equal* relational operator".

**Comment by Leonard-Reuter** [comment]
@certik I updated the proposal by adding a discussion for the major objections.

@ivan-pi I added something similar to the proposal.

**Comment by certik** [comment]
Thanks! Do you want me to merge it and do you want to submit it for the next committee meet?

**Comment by Leonard-Reuter** [comment]
@certik Yes, that'd be great =) Thank you

**Pull request merged**